File: 2026/Company/Company.Web/Modules/ServerTypes/Sales/OperationTypeColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OperationTypeRow } from "./OperationTypeRow";

export interface OperationTypeColumns {
    OperationTypeId: Column<OperationTypeRow>;
    OpCode: Column<OperationTypeRow>;
    Operation: Column<OperationTypeRow>;
}

export class OperationTypeColumns extends ColumnsBase<OperationTypeRow> {
    static readonly columnsKey = 'Sales.OperationType';
    static readonly Fields = fieldsProxy<OperationTypeColumns>();
}

File: 2026/Company/Company.Web/Modules/Sales/OperationType/OperationTypeColumns.cs
namespace Company.Sales.Columns;

[ColumnsScript("Sales.OperationType")]
[BasedOnRow(typeof(OperationTypeRow), CheckNames = true)]
public class OperationTypeColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int OperationTypeId { get; set; }
    public short OpCode { get; set; }
    [EditLink]
    public string Operation { get; set; }
}